drawBot/drawBotDrawingTools.py

```diff
@@ -462,6 +463,39 @@ def pdfImage(self):
         self._drawInContext(context)
         return context.getNSPDFDocument()
 
+    def shareImage(self, format="pdf", service="airdrop", **kwargs):
+        """
+        Share the canvas to a service with a specified format.
+
+        As default the `format` is `pdf`, any suffix drawBot supports is possible.
+
+        `service` options are `airdrop`, `mail` or `message`.
+
+        .. downloadcode:: shareImage.py
+
+            # set A4 page size
+            newPage(200, 200)
+            # draw something
+            text("Foo, bar", (10, 10))
+            # share it over airdrop
+            shareImage('pdf', service="airdrop")
+        """
+        path = tempfile.mkstemp(suffix=f".{format}")[1]
```

Comment:
Like I commented earlier, can we do this without a temp file? It's not being cleaned up, and I don't see how it could be. It's also not necessary as far as I can see: the sharing API works with all kinds of object, such as NSImage. Perhaps in-memory PDF data can be shared, too.

Reply:
A pdfDocument has no paste board support which is a requirement to send objects over. 

> The NSSharingService class is used to provide a consistent user experience when sharing items—NSURL objects, NSString objects, NSImage objects, video (through file URLs), or any object that implements the NSPasteboardWriting protocol—in macOS.

--> https://developer.apple.com/documentation/appkit/nssharingservice?language=objc

gif or mp4 has also no paste board support 

its indeed not cleaned up, as it needs to be available during the sharing process

I could add a NSSharingServiceDelegate where a callback is received when the sharing is done

https://developer.apple.com/documentation/appkit/nssharingservicedelegate?language=objc

Reply:
Then it indeed needs to be cleaned up with a delegate.

Reply:
this is silly: the delegate removes the assets to fast and fe mail can not find it anymore...

see https://gist.github.com/typemytype/ea02152e754a5b0225df2df23976afd5